### ISSUE TITLE
Reference environment variable in oidc-issuer-url

### ISSUE THREAD

**Issue by deggja** [oauth2-proxy/manifests#131]
Hello guys,

I'm curious to know if there is any way to use environment variables to reference the oidc-issuer-url in the helm chart. What I am trying to achieve is to mount a variable from a secret and reference it as such:

**alternative one (mounting tenant id from secret):**
```
oidc-issuer-url: https://login.microsoftonline.com/${tenant}/v2.0
```

**alternative two (mounting entire url from secret):**
```
oidc-issuer-url: ${url}
```

I've tried both approaches but the error message is something of this sort:

```
[2023/02/03 12:14:30] [main.go:60] ERROR: Failed to initialise OAuth2 Proxy: error intiailising provider: could not create provider data: error building OIDC ProviderVerifier: could not get verifier builder: error while discovery OIDC configuration: failed to discover OIDC configuration: unexpected status "400": {"error":"invalid_tenant","error_description":"AADSTS90002: Tenant '${tenant}' not found. Check to make sure you have the correct tenant ID and are signing into the correct cloud. Check with your subscription administrator, this may happen if there are no active subscriptions for the tenant.
```

Not sure why this does not work. I have used the same approach when defining my Azure tenant in the same config and this works exactly as expected.

**Current configuration:**
```
extraArgs:
  provider: oidc
  azure-tenant: ${tenant}  # Azure AD OAuth2 Proxy application Tenant ID
  pass-access-token: true
  email-domain: "*"
  upstream: file:///dev/null
  http-address: 0.0.0.0:4180
  oidc-issuer-url: https://login.microsoftonline.com/${tenant}/v2.0
  session-store-type: redis
extraEnv:
  - name: oidc-issuer-tenant
    valueFrom:
      secretKeyRef:
        name: oidc-issuer-url
        key: tenant
  - name: oidc_issuer_url
    valueFrom:
      secretKeyRef:
        name: oidc-issuer-url
        key: url
```

Am I going about this the wrong way? My goal is to hide the tenant ID (and other sensitive information) from the values file itself.

Thanks.

**Comment by pierluigilenoci** [comment]
Hi @deggja, 
you can mount directly a secret.

To create the secret:
```
kubectl create secret generic ${YOUR_SECRET_NAME} \
  --namespace ${NAMESPACE_OF_OAUTH2_PROXY} \
  --from-literal=client-id="${OAUTH2_APP_ID}" \
  --from-literal=client-secret="${OAUTH2_APP_PASSWORD}" \
  --from-literal=cookie-secret="${COOKIE_SECRET}"
```
or you can inject it from any vault/secret manager.

And then:
https://github.com/oauth2-proxy/manifests/blob/main/helm/oauth2-proxy/values.yaml#L18
```
config:
  existingSecret: <YOUR_SECRET_NAME>
```

Another option is to use a `configMap` to inject a config file.
Ref: 
https://github.com/oauth2-proxy/manifests/blob/main/helm/oauth2-proxy/values.yaml#L45
https://github.com/oauth2-proxy/oauth2-proxy/blob/master/contrib/oauth2-proxy.cfg.example

Does this answer your doubts?

However, the tenant ID is not strictly a secret (e.g. https://www.whatismytenantid.com/)

**Comment by deggja** [comment]
> https://www.whatismytenantid.com/

Hello @pierluigilenoci - thanks for replying.  I tried mounting the secret directly but it somehow gave me the error even then, so somehow it was not picked up.

Anyway - good point about the tenant ID not being a secret in the first place, I did not think about that.. and I think that basically revokes my need for this to work. I'll close the issue.

Thanks for pointing that out :see_no_evil:
